src/NLog/Internal/StringBuilderExt.cs

```diff
@@ -59,5 +59,99 @@ public static void Append(this StringBuilder builder, object o, LogEventInfo log
             builder.Append(Convert.ToString(o, formatProvider));
         }
 
+        /// <summary>
+        /// Appends int without using culture, and most importantly without garbage
+        /// </summary>
+        /// <param name="builder"></param>
+        /// <param name="value">value to append</param>
+        public static void AppendInvariant(this StringBuilder builder, int value)
+        {
+            // Deal with negative numbers
+            if (value < 0)
+            {
+                builder.Append('-');
+                uint uint_value = uint.MaxValue - ((uint)value) + 1; //< This is to deal with Int32.MinValue
+                AppendInvariant(builder, uint_value);
+            }
+            else
+            {
+                AppendInvariant(builder, (uint)value);
+            }
+        }
+
+        /// <summary>
+        /// Appends uint without using culture, and most importantly without garbage
+        /// 
+        /// Credits Gavin Pugh  - http://www.gavpugh.com/2010/04/01/xnac-avoiding-garbage-when-working-with-stringbuilder/
+        /// </summary>
+        /// <param name="builder"></param>
+        /// <param name="value">value to append</param>
+        public static void AppendInvariant(this StringBuilder builder, uint value)
+        {
+            if (value == 0)
+            {
+                builder.Append('0');
+                return;
+            }
+
+            // Calculate length of integer when written out
+            int length;
+            if (value >= 10000)
```

Comment:
That sounds like a premature optimization? Also we have now more less-than operations needed. My proposal is to replace the whole block with a single division (and some rounding up or down, dunno)